packages/panna-sdk/src/core/auth/siwe-auth.ts

```diff
@@ -0,0 +1,348 @@
+      this.lastChallenge = challenge;
+
+      // Create minimal LoginPayload for message signing
+      const payload: LoginPayload = {
+        address: challenge.address.toLowerCase(),
+        chain_id: challenge.chainId.toString(),
+        domain: challenge.domain,
+        uri: challenge.uri,
+        version: challenge.version,
+        nonce: challenge.nonce,
+        issued_at: challenge.issuedAt,
+        // Minimal required fields only - no expiration, resources, etc.
+        expiration_time: new Date(
+          Date.now() + DEFAULT_PAYLOAD_EXPIRATION_MS
+        ).toISOString(),
+        invalid_before: challenge.issuedAt,
+        statement: '',
+        resources: []
+      };
+
+      return payload;
+    } catch (error) {
+      console.error('Failed to generate login payload:', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Verify the signed payload and login the user
+   * This function should be called by the frontend after the user signs the message
+   */
+  public async login(params: LoginParams): Promise<boolean> {
+    try {
+      // Use the challenge response directly with signature added
+      if (!this.lastChallenge) {
+        throw new Error(
+          'No challenge available for verification. Please generate a payload first.'
+        );
+      }
+
+      const verifyRequest: AuthVerifyRequest = {
+        ...this.lastChallenge,
+        // Ensure timestamp is in ISO 8601 format with Z suffix
+        issuedAt: new Date(this.lastChallenge.issuedAt).toISOString(),
+        signature: params.signature,
+        isSafeWallet: params.isSafeWallet || false
+      };
+
+      // Verify with Panna API
+      const authResult = await pannaApiService.verifyAuth(verifyRequest);
+
+      if (authResult.token) {
+        // Store auth token and user address
+        this.authToken = authResult.token;
+        this.userAddress = authResult.address;
+
+        // Handle both expiresAt (timestamp) and expiresIn (duration in seconds) for backward compatibility
+        // Priority: expiresAt > expiresIn > null
+        this.tokenExpiresAt =
+          authResult.expiresAt || authResult.expiresIn || null;
+
+        // Store in localStorage for persistence (if available)
+        if (typeof window !== 'undefined') {
+          localStorage.setItem(STORAGE_KEYS.AUTH_TOKEN, authResult.token);
+          localStorage.setItem(STORAGE_KEYS.USER_ADDRESS, authResult.address);
+          if (this.tokenExpiresAt) {
+            localStorage.setItem(
+              STORAGE_KEYS.TOKEN_EXPIRY,
+              this.tokenExpiresAt.toString()
+            );
+          }
+        }
+
+        // Clear challenge after successful login to prevent reuse
+        this.lastChallenge = null;
+
+        return true;
+      }
+
+      console.warn('SIWE Auth - No token received from verify response');
+      return false;
+    } catch (error) {
+      console.error('Failed to verify login:', error);
```

Comment:
I think this was missed
```suggestion
      console.error('Failed to verify login:', error);
       //  Also clear challenge on error to force new challenge generation
      this.lastChallenge = null;
```